src/service/crypto-verifier-impl.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from '../log';
+import {getService} from '../service';
+import {
+  encodeByteArray,
+} from
+  '../../third_party/closure-library/encode-byte-array-trampoline-generated';
+
+
+const TAG_ = 'CryptoVerifier';
+
+const isWebkit = window.crypto && 'webkitSubtle' in window.crypto;
+const crossCrypto = isWebkit ? window.crypto.webkitSubtle :
+                               window.crypto.subtle;
+
+const VERSION = 0x00;
+
+/**
+ * A service that provides cryptographic signature verification for use
+ * within AMP.
+ */
+class CryptoVerifier {
+  /**
+   * Verifies RSA signature corresponds to the data given a list of public keys.
+   * @param {!Uint8Array} the data that was signed.
+   * @param {!Uint8Array} the RSA signature.
+   * @param {{e: !Uint8Array, n:!Uint8Array, keyHash: !Uint8Array }} the RSA
+   *     public keys. If keyHash in each public key is null, this function will
+   *     compute it.
+   * @return {!Promise<!boolean>} whether the signature is valid for one of
+   *     the public keys.
+   */
+  verify(data, signature, rsaPubKeys) {
```

Comment:
This would be more of an assert. No need to reject the promise, because it is never expected to be hit by code, so nobody has to be able to handle the error.